### PR TITLE
Replace mem::uninitialized with mem::zeroed

### LINKED ISSUE #214: Replace mem::uninitialized
See https://gankro.github.io/blah/initialize-me-maybe/, an easy global grep replace to `mem::zeroed()` or `Default` would probably suffice.

### PR DESCRIPTION
**Pull request by aloucks**
Fixes #214

### CODE REVIEW

File: ash/src/device.rs [hunk 1]
```diff
@@ -1510,7 +1510,7 @@ pub trait DeviceV1_0 {
         size: vk::DeviceSize,
         flags: vk::MemoryMapFlags,
     ) -> VkResult<*mut c_void> {
-        let mut data: *mut c_void = mem::uninitialized();
+        let mut data: *mut c_void = mem::zeroed();
```
Comment (MaikKlein) [inline ash/src/device.rs]: We should probably use `std::ptr::null_mut` here.

Comment (aloucks) [reply ash/src/device.rs]: Done!

**Comment by aloucks** [comment]
@MaikKlein Are other changes needed?

**Pull request merged**